hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -298,14 +304,69 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	}
 
 	r.Log.Info("Reconciling userdata Secret")
+	caSecret := ignitionserver.IgnitionCACertSecret(controlPlaneNamespace)
+	if err := r.Get(ctx, ctrlclient.ObjectKeyFromObject(caSecret), caSecret); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info("still waiting for ignition CA cert secret to exist")
+			return ctrl.Result{}, nil
+		} else {
+			return ctrl.Result{}, fmt.Errorf("failed to get ignition CA secret: %w", err)
+		}
+	}
+	tokenSecret := ignitionserver.IgnitionTokenSecret(controlPlaneNamespace)
+	if err := r.Get(ctx, ctrlclient.ObjectKeyFromObject(tokenSecret), tokenSecret); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.Log.Info("still waiting for ignition token secret to exist")
+			return ctrl.Result{}, nil
```

Comment:
should this be a `nodepoolCondition` `IgnitionTokenSecretAvailable` or so?

Reply:
This is accounted for in the new condition handling